examples/token-escrow/programs/token-escrow/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+pub struct EscrowCompressedTokensWithPda<'info> {
+    #[account(mut)]
+    pub signer: Signer<'info>,
+    #[account(seeds = [b"escrow".as_slice(), signer.key.to_bytes().as_slice()], bump)]
+    pub cpi_signer: AccountInfo<'info>,
+    pub compressed_token_program: Program<'info, psp_compressed_token::program::PspCompressedToken>,
+    pub compressed_pda_program: Program<'info, psp_compressed_pda::program::PspCompressedPda>,
+    pub account_compression_program:
+        Program<'info, account_compression::program::AccountCompression>,
+    pub account_compression_authority: AccountInfo<'info>,
+    pub compressed_token_cpi_authority_pda: AccountInfo<'info>,
+    pub registered_program_pda: AccountInfo<'info>,
+    pub noop_program: AccountInfo<'info>,
+    #[account(init_if_needed, seeds = [b"timelock".as_slice(), signer.key.to_bytes().as_slice()],bump, payer = signer, space = 8 + 8)]
+    pub timelock_pda: Account<'info, EscrowTimeLock>,
+    pub system_program: Program<'info, System>,
+}
+
+#[account]
+pub struct EscrowTimeLock {
+    pub slot: u64,
+}
+
+// TODO: add to light_sdk
+/// A helper function that creates a new compressed account with the change output.
+/// Input sum - Output sum = Change amount
+/// Outputs compressed account with the change amount, and owner of the compressed input accounts.
+fn create_change_output_compressed_token_account(
+    input_token_data_with_context: &[InputTokenDataWithContext],
+    output_compressed_accounts: &[TokenTransferOutputData],
+    owner: &Pubkey,
+) -> TokenTransferOutputData {
+    let input_sum = input_token_data_with_context
+        .iter()
+        .map(|account| account.amount)
+        .sum::<u64>();
+    let output_sum = output_compressed_accounts
+        .iter()
+        .map(|account| account.amount)
+        .sum::<u64>();
+    let change_amount = input_sum - output_sum;
+    TokenTransferOutputData {
+        amount: change_amount,
+        owner: *owner,
+        lamports: None,
+    }
+}
+
+#[inline(never)]
+pub fn cpi_compressed_token_transfer<'info>(
+    ctx: &Context<'_, '_, '_, 'info, EscrowCompressedTokensWithPda<'info>>,
+    proof: Option<CompressedProof>,
+    root_indices: Vec<u16>,
+    mint: Pubkey,
+    signer_is_delegate: bool,
+    input_token_data_with_context: Vec<InputTokenDataWithContext>,
+    output_compressed_accounts: Vec<TokenTransferOutputData>,
+    output_state_merkle_tree_account_indices: Vec<u8>,
+    pubkey_array: Vec<Pubkey>,
+) -> Result<()> {
+    let inputs_struct = CompressedTokenInstructionDataTransfer {
+        proof,
+        root_indices,
+        mint,
+        signer_is_delegate,
+        input_token_data_with_context,
+        output_compressed_accounts,
+        output_state_merkle_tree_account_indices,
+        pubkey_array,
+        is_compress: false,
+        compression_amount: None,
+    };
+
+    let mut inputs = Vec::new();
+    CompressedTokenInstructionDataTransfer::serialize(&inputs_struct, &mut inputs).unwrap();
```

Comment:
```suggestion
    let inputs = input_struct.try_to_vec()?;
```

But yeah, this `serialize` method is exactly what I wanted to use for the "account buffer" idea. :sweat_smile: Here it's not necessary though.